### PR TITLE
Generic Config

### PR DESCRIPTION
**Pull request by malnick**
Breaking out the configuration so we can inject enterprise functionality via init() function.

### CODE REVIEW

**Comment by malnick** [comment]
@darkonie @alberts This is ready for review

File: config/config.go [hunk 1]
```diff
@@ -42,19 +42,24 @@ type Config struct {
...
+	SegmentEvent:            "health",
+	DCOSVersion:             os.Getenv("DCOS_VERSION"),
+	DCOSClusterIDPath:       "/var/lib/dcos/cluster-id",
+	DCOSVariant:             VARIANT,
+	SignalServiceConfigPath: "/opt/mesosphere/etc/dcos-signal-config.json",
+	ExtraJSONConfigPath:     "/opt/mesosphere/etc/dcos-signal-extra.json",
+	TestFlag:                false,
+}
+
 func DefaultConfig() Config {
```
Comment (darkonie) [inline config/config.go]: you don't really need `DefaultConfig` function, just use `defaultConfig` var

File: config/config_test.go [hunk 2]
```diff
@@ -7,18 +7,13 @@ import (
...
-		SegmentEvent:            "health",
-		ClusterID:               "",
-		DCOSVersion:             "",
-		DCOSClusterIDPath:       "/var/lib/dcos/cluster-id",
-		SignalServiceConfigPath: "/opt/mesosphere/etc/dcos-signal-config.json",
-		ExtraJSONConfigPath:     "/opt/mesosphere/etc/dcos-signal-extra.json",
-		DCOSVariant:             "UNSET",
+	d := DefaultConfig()
+	if d.SegmentEvent != "health" {
+		t.Error("Expected default segment event to be health, got", d.SegmentEvent)
```
Comment (darkonie) [inline config/config_test.go]: t.Errorf("....%s", d.SegmentEvent)

File: config/config_test.go [hunk 3]
```diff
@@ -7,18 +7,13 @@ import (
...
-	expectedDefault := Config{
-		SegmentEvent:            "health",
-		ClusterID:               "",
-		DCOSVersion:             "",
-		DCOSClusterIDPath:       "/var/lib/dcos/cluster-id",
-		SignalServiceConfigPath: "/opt/mesosphere/etc/dcos-signal-config.json",
-		ExtraJSONConfigPath:     "/opt/mesosphere/etc/dcos-signal-extra.json",
-		DCOSVariant:             "UNSET",
+	d := DefaultConfig()
+	if d.SegmentEvent != "health" {
```
Comment (darkonie) [inline config/config_test.go]: it also would be nice to use constants to avoid typos

File: config/config_test.go [hunk 4]
```diff
@@ -7,18 +7,13 @@ import (
...
+		t.Error("Expected default segment event to be health, got", d.SegmentEvent)
 	}
-	gotDefault := DefaultConfig()
-	if gotDefault != expectedDefault {
-		t.Error("Expected ", expectedDefault, ", got ", gotDefault)
+
+	if d.DCOSClusterIDPath != "/var/lib/dcos/cluster-id" {
+		t.Error("Cluster ID path invalid, got", d.DCOSClusterIDPath)
```
Comment (darkonie) [inline config/config_test.go]: t.Errorf("...%s", d.DCOSClusterIDPath)

File: signal/signal.go [hunk 5]
```diff
@@ -110,7 +117,7 @@ func Start() {
 	config, configErr := config.ParseArgsReturnConfig(os.Args[1:])
 	switch {
 	case config.FlagVersion:
-		fmt.Println("DCOS Signal Service: version", VERSION, "on revision", REVISION)
+		fmt.Printf("DCOS Signal Service\n Version: %s\n Revision: %s\n DC/OS Variant: %s\n", VERSION, REVISION, config.DCOSVariant)
 		os.Exit(0)
 	default:
 		if config.Enabled == "false" {
```
Comment (darkonie) [inline signal/signal.go]: that looks odd. Can it be a bool, not string?
`if config.Enabled { ...`

Comment (malnick) [inline signal/signal.go]: This is an artifact from gen lib writing this file. I forget the actual issue we had, but we need to make this a string :/

File: config/config_test.go (hunk 2 above)
Comment (malnick) [inline config/config_test.go]: We can leave these here since `testing` lib logger acts like `.Println` and adds the spaces for you.

**Pull request merged**